[WIP] Initial setup for the page builder.

anuragteapot opened the pull request:
### Summary of Changes
This PR introduces the new feature Page Builder. With Apodis, the new frontend template, users are able to define the structure of their website with an editor. It allows to drag and drop the elements grid, column, container, module position, message and component into the order and position the user wants. These elements are plugins and therefore expandable and (de-) activatable.

### Testing Instructions
Log in to the Joomla backend and go to System -> Site Template Styles.
Set Apodis as the default frontend template and look into its settings.
Here you should find the Page Builder tab and the editor.
Use the editor, try it out, add, edit and remove elements. Save and reload. Change the target device. Test module positions like 'sidebar-right' or 'footer' and assign modules (outside of the Page Builder).
Look at the frontend. Does Apodis look like you expected with the elements set in the editor? Have the settings of each element, e.g. classes or gaps, the correct effect? Is it responsive and adjust on different screen sizes? Are there any browser console errors caused by JavaScript?

### Documentation Changes Required
Yes. You can find it in the [Joomla Documentation](https://docs.joomla.org/J4.x:Page_Builder).


This is draft Pull request.

**Don't merge this PR this should be created by students.**

anuragteapot commented on administrator/components/com_templates/resources/scripts/components/app.vue:
```diff
@@ -0,0 +1,377 @@
+      <h2>{{ translate('COM_TEMPLATES_SETTINGS') }}</h2>
+      <hr>
+      <form>
+        <!-- Settings for editing positions -->
```
Split this file in small components.

anuragteapot commented on administrator/components/com_templates/resources/scripts/plugins/translate.js:
```diff
@@ -0,0 +1,43 @@
+	return Joomla.JText._(key, key);
+};
+
+
```
Remove extra line.

anurento commented on administrator/components/com_templates/resources/scripts/components/app.vue:
```diff
@@ -0,0 +1,378 @@
+              });
+            }
+            else {
+              alert('Grid size cannot be greter than 12');
```
Use Joomla alert.

anuragteapot replied:
use Joomla alert.  https://github.com/joomla-projects/gsoc19_page-builder/blob/01e29b55827ae991376fbe9b7ba399c72521a81f/administrator/components/com_media/resources/scripts/components/app.vue#L52

anuragteapot commented on administrator/components/com_templates/resources/scripts/components/app.vue:
```diff
@@ -0,0 +1,378 @@
+              });
+            }
+            else {
+              alert('Grid size cannot be greater than 12');
```
Use Joomla alert https://github.com/joomla-projects/gsoc19_page-builder/blob/01e29b55827ae991376fbe9b7ba399c72521a81f/administrator/components/com_media/resources/scripts/components/app.vue#L52

anuragteapot commented on administrator/components/com_templates/resources/styles/components/_app.scss:
```diff
@@ -0,0 +1,67 @@
+// App.vue
+
+#pageBuilder {
```
I think class is better than id. 

`.pageBuilder`

anuragteapot commented on administrator/components/com_templates/resources/styles/components/_app.scss:
```diff
@@ -0,0 +1,67 @@
+}
+
+#pageBuilder {
+
```
Same class and remove extra line.

anuragteapot commented on administrator/components/com_templates/resources/styles/components/_app.scss:
```diff
@@ -0,0 +1,67 @@
+    }
+}
+
+#newgrid .modal-body img {
```
class

bahl24 replied:
To fire up a modal, id is required to link the button to the modal. Anyways, I am working to adopt the modal component system as in com_media.

anuragteapot commented on administrator/components/com_templates/resources/styles/components/_rawDisplayer.scss:
```diff
@@ -0,0 +1,8 @@
+// RawDisplayer.vue
+
+.raw {
+
```
Extra line

anuragteapot commented on administrator/components/com_templates/resources/scripts/components/app.vue:
```diff
@@ -0,0 +1,378 @@
+          this.showSettings = true;
+        }
+      },
+      back() {
```
If `back()` function is something like reset then i would suggest you to change it to `reset()`

anuragteapot commented on administrator/components/com_templates/public/index.html:
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
```
Is any need of this file ?

anuragteapot commented:
@bahl24 @fancyFranci I think to install template you need  to add data to sql file https://github.com/joomla-projects/gsoc19_page-builder/blob/66d6d3214f126e0d3883a3387916d57a20aa1db8/installation/sql/mysql/joomla.sql#L709  Like this one.

anuragteapot commented:
@bahl24 @fancyFranci Why so many changes ?

bahl24 commented:
> @bahl24 @fancyFranci Why so many changes ?

The earlier drone hash was expired, so drone tests were not running. I've updated the hash now

fancyFranci commented:
> 
> @bahl24 @fancyFranci Why so many changes ?

Most of them are moving files like discussed or from the 4.0 branch. Which changes do you mean exactly?

anuragteapot commented on build/media_source/com_templates/vue/components/settings/editgrid.vue:
```diff
@@ -0,0 +1,40 @@
+    methods: {
+        editGrid() {
+            if (this.grid_class !== '') {
+                this.grid.options.class = this.grid_class;
```
Why we are updating props in child component ?  You are taking advantage of the weakness of Vue. ;P   

This is a very bad practice. This code gets breaks if something change in Vue and its also breaks your pagebuilder.

Read about one way data binding. https://vuejs.org/v2/guide/components-props.html#One-Way-Data-Flow

anuragteapot commented on build/media_source/com_templates/vue/components/settings/editgrid.vue:
```diff
@@ -0,0 +1,40 @@
+                this.grid.options.class = this.grid_class;
+            }
+            this.grid_class = '';
+            this.$emit('reset');
```
If you want to pass data to the parent component, pass in event. 

But, the best solution is Vuex read about, vue without vuex is incomplete.

anuragteapot commented on build/media_source/com_templates/vue/components/settings/addcolumn.vue:
```diff
@@ -0,0 +1,60 @@
+    methods: {
+        addColumn() {
+            if (this.column_size !== '') {
+                this.grid.children.push({
```
Same here.

anuragteapot commented on build/media_source/com_templates/vue/pagebuilder.js:
```diff
@@ -0,0 +1,34 @@
+
+// Add the plugins
+Vue.use(Translate);
+Vue.use(VueJsModal);
```
Why we are using VueJsModal we already have Vuetify .?

fancyFranci replied:
Vuetify came after VueJSModal and we will remove Vuetify again.

anuragteapot commented on build/media_source/com_templates/vue/pagebuilder.js:
```diff
@@ -0,0 +1,34 @@
+// Add the plugins
+Vue.use(Translate);
+Vue.use(VueJsModal);
+Vue.use(Vuetify);
```
If we are using Vuetify only for sidebar you can create own and remove vuetify.

fancyFranci replied:
Nitish did already created an alternative to vuetify.

anuragteapot commented:
![Screenshot from 2019-06-28 21-37-27](https://user-images.githubusercontent.com/30978328/60355817-f7309a80-99ec-11e9-9c24-6995fe1d9c99.png)


Error while adding custom class .

anuragteapot commented on build/media_source/com_templates/scss/pagebuilder.scss:
```diff
@@ -0,0 +1,136 @@
+
+    .btn-wrapper {
+      visibility: hidden;
+      position: absolute;
```
```
    visibility: hidden;
    position: absolute;
    top: -32px;
    right: 0;
    background-color: #9898ea;
}
```

anuragteapot replied:
If you make this change this will looks something like 
![Screenshot from 2019-06-28 22-03-53](https://user-images.githubusercontent.com/30978328/60357242-c05c8380-99f0-11e9-9934-8f8f9368aa5f.png)

anuragteapot replied:
If its looks good then  You can change the background color you want.

fancyFranci replied:
Done. Thanks for the draft!

anuragteapot commented on build/media_source/com_templates/vue/components/app.vue:
```diff
@@ -0,0 +1,172 @@
+						<span class="sr-only">{{ translate('COM_TEMPLATES_EDIT_GRID') }}</span>
+					</button>
+					<button type="button" class="btn btn-lg" @click="deleteGrid(grid)">
+						<span class="icon-cancel"></span>
```
`icon-cancel`   change it to delete icon. Because, you are not canceling any think you are deleting.

anuragteapot replied:
and  try to make icons small

fancyFranci replied:
This is a class from fontawesome which Joomla is using to display that icon.
It was an issue that the icons were too small. People could have problems with small icons.

anuragteapot commented on build/media_source/com_templates/vue/components/grid-element.vue:
```diff
@@ -0,0 +1,154 @@
+				</button>
+				<button type="button" class="btn btn-lg" @click="deleteColumn(column)">
+					<span class="icon-cancel"></span>
+					<span class="sr-only">{{ translate('COM_TEMPLATES_DELETE_COLUMN') }}</span>
```
Delete icon.

zero-24 commented:
@Anu1601CS @fancyFranci when you get a minute can you update your branch to the latest https://github.com/joomla-projects/gsoc19_page-builder/tree/4.0-dev that should fix the first error issued by drone and the actual tests can run.

Please do not use joomla/joomla-cms:4.0-dev as source as there we have a different drone signature. Please always use joomla-projects/gsoc19_page-builder:4.0-dev

anuragteapot commented on administrator/components/com_templates/tmpl/style/edit.php:
```diff
@@ -99,6 +99,10 @@
 			<?php echo HTMLHelper::_('uitab.endTab'); ?>
 		<?php endif; ?>
 
+		<?php echo HTMLHelper::_('uitab.addTab', 'myTab', 'pagebuilder', Text::_('COM_TEMPLATES_PAGE_BUILDER')); ?>
```
CS

anuragteapot commented on build/media_source/com_templates/vue/components/app.vue:
```diff
@@ -0,0 +1,92 @@
+        this.$modal.show('add-element');
+      },
+	  deleteElement(element) {
+        this.$store.commit('deleteElement', {element: element});
```
you can also write like this and CS

`this.$store.commit('deleteElement', { element })`

anuragteapot commented on build/media_source/com_templates/vue/components/elements/item.vue:
```diff
@@ -0,0 +1,75 @@
+
+			<grid v-if="element.type === 'Grid'" :grid="element"></grid>
+
+			<item v-else v-for="child in element.children" :item="child" @delete="deleteElement(child)"></item>
```
Provide  key props in for loop

anuragteapot replied:
So, when an update comes it can identify.

bahl24 replied:
@Anu1601CS  I don't think we have some unique identifier which can be set as a key prop, as our JSON only has a type, options and children. So can we implement without a key or should we add some key/counter externally?

anuragteapot commented on build/media_source/com_templates/vue/components/modals/modal-add-element.vue:
```diff
@@ -0,0 +1,105 @@
+			<div class="row">
+				<div class="col-4">
+					<div class="list-group" id="list-tab" role="tablist">
+						<li v-for="element in allowedChildren" data-toggle="list" role="tabpanel"
```
Key props

anuragteapot commented on build/media_source/com_templates/vue/components/modals/modal-add-element.vue:
```diff
@@ -0,0 +1,105 @@
+					</div>
+				</div>
+				<div class="col-8 tab-content">
+					<div v-for="element in allowedChildren" :id="'list-'+element.name" class="tab-pane fade" role="tabpanel">
```
Key props

anuragteapot commented on build/media_source/com_templates/vue/components/elements/grid.vue:
```diff
@@ -0,0 +1,310 @@
+<template>
+	<grid-layout :layout="allItems"
+				 :col-num="gridSize"
+				 :is-draggable="true"
```
CS which prettier you guys are using  ?

anuragteapot commented on plugins/pagebuilder/grid/grid.php:
```diff
@@ -0,0 +1,83 @@
+
+		return array(
+			'name'     		=> Text::_('PLG_PAGEBUILDER_GRID_NAME'),
+			'description' 	=> Text::_('PLG_PAGEBUILDER_GRID_DESC'),
```
CS

anuragteapot commented on build/media_source/com_templates/vue/components/elements/item.vue:
```diff
@@ -0,0 +1,76 @@
+<template>
+	<div :class="['item', element.type, element.options.class]">
+		<div class="btn-wrapper btn-group">
+			<button type="button" class="btn btn-primary" @click="editElement(element)">
```
Make button small add `btn-sm`

anuragteapot commented on build/media_source/com_templates/vue/components/elements/item.vue:
```diff
@@ -0,0 +1,76 @@
+				<span class="icon-options"></span>
+				<span class="sr-only">{{ translate('COM_TEMPLATES_EDIT') }}</span>
+			</button>
+			<button type="button" class="btn btn-danger" @click="$emit('delete')">
```
Make button small add `btn-sm`

anuragteapot commented on build/media_source/com_templates/scss/pagebuilder.scss:
```diff
@@ -0,0 +1,197 @@
+  .btn-wrapper {
+    visibility: hidden;
+    position: absolute;
+    top: -1.5rem;
```
Make `top : -2rem`

anuragteapot commented on build/media_source/com_templates/vue/components/settings/editelement.vue:
```diff
@@ -0,0 +1,157 @@
+        <fieldset>
+            <legend>{{ translate('COM_TEMPLATES_EDIT') }}{{ elementSelected.type }}</legend>
+            <div class="form-group">
+                <label for="element_class">{{ translate('COM_TEMPLATES_ADD_CLASS') }}</label>
```
Use Joomla input field class

anuragteapot commented:
@bahl24 @fancyFranci Make sidebar better using padding and margin.

anuragteapot commented on build/media_source/com_templates/vue/components/settings/editelement.vue:
```diff
@@ -0,0 +1,157 @@
+                            <tr>
+                                <th scope="row"><i class="fas fa-mobile-alt fa-2x"></i></th>
+                                <td class="form-group">
+                                    <select name="element_offset" id="element_offset" class="form-control form-control-sm" v-model="element_offset.xs">
```
Joomla select style

anuragteapot commented on build/media_source/com_templates/vue/components/settings/editelement.vue:
```diff
@@ -0,0 +1,157 @@
+                            <tr>
+                                <th scope="row"><i class="fas fa-tablet-alt fa-2x"></i></th>
+                                <td class="form-group">
+                                    <select name="element_offset" id="element_offset" class="form-control form-control-sm" v-model="element_offset.sm">
```
Joomla select style

anuragteapot commented:
When user click on the setting of column sidebar should open with the previous value of that column.

anuragteapot commented on build/media_source/com_templates/vue/components/elements/grid.vue:
```diff
@@ -0,0 +1,312 @@
+          occupied = this.atPosition(position, y);
+        }
+
+        return {x: x, y: y, w: w};
```
try this `return { x, y, w }`;

anuragteapot commented on administrator/components/com_templates/Helper/RenderHelper.php:
```diff
@@ -0,0 +1,124 @@
+
+		PluginHelper::importPlugin('pagebuilder');
+
+		$html = '<div class="container-fluid">' . self::render($elements) . '</div>';
```
is container is always going to be fluid. ? @bembelimen

anuragteapot commented on build/media_source/com_templates/vue/components/app.vue:
```diff
@@ -0,0 +1,110 @@
+        <li class="nav-item" @click="deviceWidth = '1050px'">
+          <a class="nav-link" data-toggle="pill" role="tab"><i class="fas fa-laptop fa-lg"></i></a>
+        </li>
+        <li class="nav-item" @click="deviceWidth = '1250px'">
```
For desktop can we make width `100%`

anuragteapot commented:
@bahl24 @fancyFranci Use auto-indent in your IDE to fix all indentation.

anuragteapot commented:
Oops sorry.

anuragteapot commented on build/media_source/com_templates/vue/components/settings/editelement.vue:
```diff
@@ -0,0 +1,203 @@
+            deep: true,
+            handler() {
+                this.element_class = this.elementSelected.options.class;
+                this.element_module_chrome = this.elementSelected.options.module_chrome ? this.elementSelected.options.module_chrome : 'none';
```
Are you sure that `this.elementSelected.options.class` value always be present? If options are not present then it will throw error.

bahl24 replied:
We have added "class" in JSON corresponding to every element

wilsonge committed:
[4.0] Correcting frontend login/registration menu items password label (#25533)

SharkyKZ committed:
Rename class to match file name (#25617)

bembelimen committed:
Remove dynamic SVG colouring.

Hackwar committed:
Updating branch (#418)

Updating branch

fancyFranci committed:
Remove pagebuilder parameter from Cassiopeia

fancyFranci committed:
Set sm as default device (mobile-first like Bootstrap)

fancyFranci committed:
Change layout depending on device

Hackwar committed:
Merge branch '4.0-dev' of https://github.com/joomla/joomla-cms into release

Hackwar committed:
Updating composer.lock

bahl24 committed:
component can now be dragged only to allowed loc

bahl24 committed:
Merge branch 'editor' of https://github.com/joomla-projects/gsoc19_page-builder into editor

Hackwar committed:
Updating to latest 4.0-dev (#419)

Updating to latest 4.0-dev

bahl24 committed:
ensure comp and mess are not in same elem

Hackwar committed:
Quickicons clientid (#415)

Quickicons clientid

fancyFranci committed:
Sidebar gets white background when scrolling

bahl24 committed:
add true value in options in concerned elem

fancyFranci committed:
Refactor template.js

bahl24 committed:
retrieve comp & mess on saving and reopening

fancyFranci committed:
Add descriptions to device selection

fancyFranci committed:
Add xl as device size

wilsonge committed:
Remove the colour variable

bembelimen committed:
Merge pull request #420 from joomla/fix-css-color-names

Remove the colour variable

fancyFranci committed:
Make add-element more universal (step to move specific data to plugins)

fancyFranci committed:
Make devices wider to fix break on open settings

fancyFranci committed:
Add xl offset to settings

Hackwar committed:
Fixing selector for toolbars in systemtests for com_media

fancyFranci committed:
Shift configuration into plugin (moduleposition first)

fancyFranci committed:
Plugin determines which options get displayed on editor item

zero-24 committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into 4.0-dev

bahl24 committed:
make component and message flexible with plugins

bahl24 committed:
merge editor into local branch

bahl24 committed:
solve offset error

fancyFranci committed:
Generate offset classes in plugin renderer

fancyFranci committed:
Set selected device like stored activeDevice

anuragteapot committed:
Merge branch '4.0-dev' into editor

anuragteapot commented on build/media_source/com_templates/vue/components/app.vue:
```diff
@@ -0,0 +1,141 @@
+      this.checkAllowedElements();
+    },
+    mounted() {
+      if(document.getElementsByClassName('drag_component').length) {
```
Is `$refs` is not working  ?

anuragteapot commented on build/media_source/com_templates/vue/components/app.vue:
```diff
@@ -0,0 +1,141 @@
+          element.__vue__.$data.element.options.message = false;
+        }
+        element.classList.remove('drag_' + location);
+        document.getElementById('placeholder_' + location).appendChild(document.getElementById('drag_' + location));
```
can we use `$refs` ?

bahl24 replied:
I think it will result in code duplication as we cant use something like `'placeholder_' + location` with $refs, we would have to write it twice `placeholder_component` & `placeholder_message`

Hackwar committed:
Fixing selector for toolbars in systemtests for com_media (#421)

Fixing selector for toolbars in systemtests for com_media

fancyFranci committed:
Try to make nicer apodis styling

fancyFranci committed:
Update sizes when switching devices

fancyFranci committed:
Add component and message to frontend rendering

puneet0191 committed:
Fixing Failing Tests

bembelimen committed:
Merge pull request #32 from puneet0191/patch-1

Fixing Failing Tests

puneet0191 committed:
Fixing Duplicate Image Issue

bembelimen committed:
Merge pull request #33 from puneet0191/patch-2

Fixing Duplicate Image Issue

Hackwar committed:
Clearing file input in com_media for tests

chmst committed:
Reset changes in com_modules, set user state ih quickicons

Hackwar committed:
Clearing file input in com_media for tests (#422)

Adding yet another wait in system tests

bembelimen committed:
Merge branch 'backend-template' of github.com:bembelimen/joomla-cms into release

# Conflicts:
#	tests/Codeception/acceptance/administrator/components/com_media/MediaListCest.php

chmst committed:
Fix load the filter state

chmst committed:
redo changes

bembelimen committed:
Remove unnessecary lines

bembelimen committed:
Merge pull request #423 from joomla/module-select-clientid

set Client_id for modules

bembelimen committed:
Fix typo

Hackwar committed:
Downgrade choco to a version that works (#25804)

wilsonge committed:
Merge branch '4.0-dev' into backend-template

bembelimen committed:
Update tests/Codeception/acceptance/administrator/components/com_media/MediaListCest.php

Co-Authored-By: Richard Fath <richard67@users.noreply.github.com>

bembelimen committed:
Remove variables import

richard67 committed:
Real null dates for mod_latestactions and mod_privacy_dashboard (#25805)

richard67 committed:
Fix schema updates (#34)

Fix schema updates for consistent use of null values in module table and correct names quoting for PostgreSQL.

bembelimen committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into release

# Conflicts:
#	installation/sql/mysql/joomla.sql
#	installation/sql/postgresql/joomla.sql

brianteeman committed:
[4.0] Action log date (#25806)

This PR makes the data column here display in the same way as in Privacy Requests.

brianteeman committed:
[4.0] htaccess.txt clarifications (#25737)

Updated the comments in the htaccess file to make it clearer that the modules referred to are Apache modules and not Joomla modules

(I just had someone asking why joomla didnt ship with mod_rewrite installed like all the other modules)

zero-24 committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into 4.0-dev

infograf768 committed:
Merge branch '4.0-dev' into backend-template

chmst committed:
Add missing language strings for view help

bembelimen committed:
Merge pull request #424 from joomla/add-language-strings-help

Add missing language strings for view help

richard67 committed:
Correct sequence numbers for PostgreSQL (#35)

Commit:
Fix output variable (#25815)

brianteeman committed:
[4.0] Version History compare a11y (#25818)

bahl24 committed:
fix arbitrary deletion of new cols

bahl24 committed:
add vuex-persistedstate to store activeDevice

bahl24 committed:
fix drone npm error

bahl24 committed:
WIP add BS in scss and js file

richard67 committed:
Update script.php (#40)

richard67 committed:
More corrections for schema updates (#37)

* Correct schema update for PostgreSQL

* Align schema update with joomla.sql for MySQL

* Align schema update with joomla.sql for PostgreSQL

richard67 committed:
Consistent use of null for modules table in joomla.sql (#38)

richard67 committed:
Update ExtensionHelper.php (#39)

bembelimen committed:
Merge branch 'backend-template' of github.com:bembelimen/joomla-cms into release

Commit:
Fix typo

bembelimen committed:
Merge pull request #426 from Quy/patch-1

Fix typo

SharkyKZ committed:
Rename class to match file name (#25830)

brianteeman committed:
[4.0] Action logs filters (#25825)

brianteeman committed:
[4.0] com_privacy filters (#25824)

Make sure that all the filters have a label only

Commit:
[4.0] Fix markup (#25832)

brianteeman committed:
[4.0] Double translation of template preview  button (#25821)

The text for the button was being translated twice

To test enable language debug

Before this PR the button will display as `??**COM_TEMPLATES_BUTTON_PREVIEW**??`

After the PR it will be correctly displayed as `**COM_TEMPLATES_BUTTON_PREVIEW**`

martyf committed:
[4.0] Added additional TinyMCE 5 configuration option for "external_plugins" (#25734)

Added support for external plugins. This has been done using a repeatable pairing of plugin names and paths. See “External Plugins” tab in Plugin settings.

This was done outside of the main options so that they can be globally applied to all configuration sets.

zero-24 committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into 4.0-dev

bembelimen committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into release

# Conflicts:
#	administrator/templates/atum/html/layouts/chromes/body.php
#	administrator/templates/atum/html/layouts/chromes/well.php

bembelimen committed:
Merge branch 'release' of github.com:joomla/backend-template into release

bembelimen committed:
Merge pull request #427 from joomla/subhead-scroll

Subhead scroll

bembelimen committed:
Fix broken JS scroller class

bembelimen committed:
Merge branch '4.0-dev' into backend-template

fancyFranci committed:
Create pagebuilder field

fancyFranci committed:
Set pagebuilder field in Apodis settings

fancyFranci committed:
Move pagebuilder files from com_templates into system/fields

fancyFranci committed:
Adjust webpack and vue files and separate state for script options

fancyFranci committed:
Fix add-grid image paths

fancyFranci committed:
Fix json value for pagebuilder field

fancyFranci committed:
Remove pagebuilder from com_templates

fancyFranci committed:
Get next smaller device size when element has no size for current one

Commit:
Merge remote-tracking branch 'origin/4.0-dev' into editor

Commit:
Move language strings from com_templates into JLib

Commit:
Fix loading of empty value in field

bahl24 committed:
shift to scss files for bs4

bahl24 committed:
prevent shifting of editor to left on closing nav

brianteeman committed:
[4.0] Redirect plugin (#25846)

wilsonge committed:
Merge pull request #25570 from bembelimen/backend-template

[4.0] Backend template

dgrammatiko committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into 4.0-dev-spinner-nth-attempt

brianteeman committed:
[4.0] Update atum template preview

Simple PR to update the preview images for the Atum template

wilsonge committed:
Merge pull request #25857 from brianteeman/template2

[4.0] Update atum template preview

brianteeman committed:
[4.0] Dashboard links (#25853)

The text of the new dashboard links is currently "Dashboard: Content"

It is better english to say "Content Dashboard".

This PR also removes the :

When a screen reader sees a : it inserts a long pause and as this text is really just for screen readers it makes it confusing

alikon committed:
[4.0][com_finder] wrong update join syntax for postgresql (#25859)

wilsonge committed:
Add back new line for phpcs

brianteeman committed:
[4.0] Checkboxes field (#25842)

Change the field layout so that the `<legend>` is set to be sr-only and that the field label is rendered.

Commit:
Add description for selected device

Commit:
Fix styling of active device

Commit:
Generate device toggler

Commit:
Set min-width and max-width to editor for each device

bahl24 committed:
prevent having empty fields

bahl24 committed:
display module_chrome in item

bahl24 committed:
fix module pos field in settings is empty

bahl24 committed:
prevent empty mod chrome in sidebar

bahl24 committed:
avoid code repetition

fancyFranci committed:
Resize beyond grid border

fancyFranci committed:
Create invisible column as row placeholder to enhance grid directly

fancyFranci committed:
Fix bug where big columns did not move to the next row after resizing

SharkyKZ committed:
Fix tag edit layout (#25880)

fancyFranci committed:
Change device texts

SharkyKZ committed:
Move admin tags view metadata file (#25881)

brianteeman committed:
[4.0] Add dashboard module (#25871)

Maybe its just me but I was looking for a specific dashboard module. I think better text is Add module to the dashboard

dgrammatiko committed:
Merge branch '4.0-dev' into spinner-nth-attempt

anuragteapot commented on administrator/components/com_templates/Helper/RenderHelper.php:
```diff
@@ -0,0 +1,123 @@
+/**
+ * Template Helper class.
+ *
+ * @since  4.0
```
`@since  __DEPLOY_VERSION__` Because we don't know when this feature merge in Joomla.

anuragteapot commented on administrator/components/com_templates/Helper/RenderHelper.php:
```diff
@@ -0,0 +1,123 @@
+	 *
+	 * @return  string
+	 *
+	 * @since   4.0
```
`@since  __DEPLOY_VERSION__` Because we don't know when this feature merge in Joomla.

anuragteapot commented on administrator/components/com_templates/Helper/RenderHelper.php:
```diff
@@ -0,0 +1,123 @@
+	 *
+	 * @return  array|boolean
+	 *
+	 * @since   4.0
```
`@since  __DEPLOY_VERSION__` Because we don't know when this feature merge in Joomla.

anuragteapot commented on administrator/components/com_templates/Helper/RenderHelper.php:
```diff
@@ -0,0 +1,123 @@
+	 *
+	 * @return  string
+	 *
+	 * @since   4.0
```
`@since  __DEPLOY_VERSION__` Because we don't know when this feature merge in Joomla.

infograf768 committed:
[4.0][Backend Template] Correcting plugin options duplication (#25913)

brianteeman committed:
[4.0] a website menu is not role="menu" (#25909)

I know it seems obvious to put role=menu, role=menuitem on our main menu but it is completely wrong. It is primarily for mimicking native OS menus not for navigation menus.

"Don’t use ARIA menu semantics in navigation menu systems." [inclusive design(https://inclusive-components.design/menus-menu-buttons/)]

Also it is wrong to have aria-haspopup to indicate the presence of a collapsed submenu - the correct attribute is aria-expanded=true/false

"Use aria-expanded to indicate the open/closed state of a button-activated navigation menu." [inclusive design(https://inclusive-components.design/menus-menu-buttons/)]

wilsonge committed:
Remove redundant SCSS

wilsonge committed:
Tag Joomla 4 Alpha 11

richard67 committed:
Remove duplicate sessiongc from core extensions list (#25918)

infograf768 committed:
[4.0][Backend Template] Correcting custom admin menu items edit display (#25908)

brianteeman committed:
[4.0] Dashboard admin submenu modules (#25901)

The markup for these modules is incorrect. The h2 heading should only contain the title not the title plus dropdown menu.

In particular this causes issues for assistive technology as it doesn't understand what the title is. You can test this easily using the skip-to menu (hit tab) and you will see that the h2 has been truncated to nothing.

Apply this PR and its all good

wilsonge committed:
Fix switcher text alignment in RTL

wilsonge committed:
Tweak padding in backend template to fix toolbar button alignment

wilsonge committed:
Improve menu dropdown in RTL

infograf768 committed:
[4.0][Backend Template]Increasing admin menu width to let longer menu (#25922)

titles display correctly with existing word-break

infograf768 committed:
[4.0] Adding Read More link (#25907)

richard67 committed:
[4.0] Update manifest cache after testing sampledata plugin installation (#25919)

SharkyKZ committed:
Namespace JFormField uses (#25514)

SharkyKZ committed:
Namespace JFormRule uses (#25515)

SharkyKZ committed:
Directory tree icons not clickable (#25836)

bahl24 committed:
[4.0] Featured view: corrected rss feed func (#24028)

SharkyKZ committed:
[4.0] Access library prepared statements (#25917)

infograf768 committed:
[4.0][backend template] Create opacity for disabled top menu items (#25906)

* [4.0][backend template] Create opacity for disabled top menu items

* Adding pointer-events

SharkyKZ committed:
Application library prepared statements #25926

infograf768 committed:
[4.0][Admin template] Correcting config permissions and text filters
display

bahl24 committed:
fix #40 - removing elem removes comp/mess comp

HLeithner committed:
Correcting config permissions and text filters display #25934

bahl24 committed:
fix #43 now new col relate to activeDevice

brianteeman committed:
[4.0] Disabled buttons (#25933)

PR for #25874t
Apply Cursor not allowed to the searchtools buttons to be consistent with the toolbar buttons

wilsonge committed:
Reset back to dev

brianteeman committed:
[4.0] Remove sample data step from install (#25828)

PR for #25813

SharkyKZ committed:
[4.0] Add validation capability to access level field (#25817)

bahl24 committed:
correct col size rendering in frontend

bahl24 committed:
fix #39 module position not visible in frontend

bahl24 committed:
fix #39 module chrome rendering

wilsonge committed:
Fix switcher RTL in body areas (e.g. com_modules)

brianteeman committed:
[4.0] Back to Dashboard string

the string TPL_BACK_TO_CONTROL_PANEL was removed in error in the template repo. This PR puts it back and at the same ttime takes the opportunity to name it correctly as TPL_ATUM_BACK_TO_CONTROL_PANEL

The string is used for the link on the logo in the top corner

infograf768 committed:
[4.0][Admin menu] Improving various aspects including multilingual (#25929)

wilsonge committed:
Namespace table interface and add missing import to interface

fancyFranci committed:
Resizing over grid border sets width depending on the drag distance

Hackwar committed:
Switching docker images for drone to Joomla-own

HLeithner committed:
Switching docker images for drone to Joomla-own #25936

zero-24 committed:
update branch and drone signature

rdeutz committed:
Merge branch '4.0-dev' into mon6

alikon committed:
[4.0][admin menu] - fix wrong group by in preset menu (#25938)

* fix wrong group by

* fix wrong group by

* fix wrong group by

* lang_code

* lang_code

* lang_code

brianteeman committed:
[4.0] composer update (#25940)

wilsonge committed:
Merge branch '4.0-dev' into mon6

wilsonge committed:
[4.0] Back to Dashboard string (#25935)

the string TPL_BACK_TO_CONTROL_PANEL was removed in error in the template repo. This PR puts it back and at the same ttime takes the opportunity to name it correctly as TPL_ATUM_BACK_TO_CONTROL_PANEL

The string is used for the link on the logo in the top corner

dgrammatiko committed:
Merge branch '4.0-dev' into spinner-nth-attempt

HLeithner committed:
Prepared plg system remember (#25539)

fancyFranci committed:
Refactor ordering of columns

fancyFranci committed:
Fix saving changed column order

fancyFranci committed:
Fix column positions on device change

infograf768 committed:
[admin template] Missing mod_menu lang strings for modern preset (#25956)

bahl24 committed:
change @since version

bahl24 committed:
fix #40 to account for future generation

bembelimen committed:
Implement base parameter for font, background and link color
Clean up assets with loading the core bootstrap
Removed unused css
Added language files

fancyFranci committed:
Merge remote-tracking branch 'origin/4.0-dev' into template-test

# Conflicts:
#	build/build-modules-js/compilecss.es6.js
#	package-lock.json

bahl24 committed:
take care of exceptions & unused methods

fancyFranci committed:
Merge remote-tracking branch 'origin/editor' into template-test

bembelimen committed:
Add context to plugin call
Rename plugin call to a more specific version

bembelimen committed:
Remove container enhancements

fancyFranci committed:
Refactor scss and adjust to new backend template

fancyFranci committed:
Changed drag items styling and show remove icon when its available only

fancyFranci committed:
Make item classes more readable

fancyFranci committed:
Enhance height of columns by children

bahl24 committed:
try fixing drone npm err

brianteeman committed:
System Dashboard Install Languages (#25983)

brianteeman committed:
[4.0] System dashboard strings (#25984)

* Information warnings

Warnings is now in the Information section. Update the page title to show Information: Warnings instead of Extensions: Warning and remove unused string

* Database is now in the information section - not extensions

brianteeman committed:
Maintain group on system dashboard (#25985)

Renamed to Maintenance to be consistent with the linked pages

brianteeman committed:
Change the required field message to something friendlier and the same as the browser default message (#25986)

brianteeman committed:
Update Sites dashboard (#25988)

Change the string Update Sources on the dashboard to Update Sites to match the rest of Joomla

SharkyKZ committed:
Namespace JUri and JRouter uses (#25992)

brianteeman committed:
[4.0] Remove menuitem letter-spacing (#25991)

Especially as it can cause undesired issues with non latin characters or when the user has _zoomed_ on the page to increase the text size for accessibility. The font designers are pretty good and know more about typography than we do.

brianteeman committed:
The update table is not up to date link (#25982)

There is no good reason for this link to open in a new page

fancyFranci committed:
Reset height on device change

bahl24 committed:
merge configs with element.options

bahl24 committed:
Merge branch 'editor' of https://github.com/joomla-projects/gsoc19_page-builder into editor

bahl24 committed:
return data->options to the php render method

fancyFranci committed:
Merge pull request #45 from joomla-projects/parameters

Parameters, Plugin-Context, Assets

fancyFranci committed:
Fix Apodis description in language file

SharkyKZ committed:
Prepared statements (#25977)

HLeithner committed:
[4.0] Convert plugin system updatenotification to prepared statements (#25542)

* Convert plugin system updatenotification to prepared statements

* Fix missing variable replacement

SharkyKZ committed:
[4.0] User library prepared statements (#25968)

* Prepared statements

* Use setLimit()

* Align

* Accidentally removed

* Use one call

brianteeman committed:
[4.0] Enable com_finder (#25839)

Prior to Joomla 4 we had two search options - we now just have finder aka smart search. So it makes no sense for this to be disabled on NEW installs. This PR enables the plugin on NEW installs so that you know longer get the warning etc.

This does not change EXISTING sites

fancyFranci committed:
Integrate container class container-fluid

fancyFranci committed:
Correct comments and return values in plugins

fancyFranci committed:
Add language string for container config fluid

Hackwar committed:
Fix catid (catid 1 is root)

Hackwar committed:
Updating joomla-projects/joomla-browser

wilsonge committed:
Merge pull request #26016 from Hackwar/patch-20

[4.0] Fix catid (catid 1 is root)

brianteeman committed:
User select button (#26026)

brianteeman committed:
Alt text (#26019)

All the field labels are "Alt Text" except for this one

brianteeman committed:
LTR RTL Direction com_newsfeeds (#26023)

Removes unnucessary word

brianteeman committed:
edit banner missing div (#26021)

The publishing fields are missing a tab which reslts in the label and input being misaligned.

brianteeman committed:
com_tags options (#26020)

Fix the css so that the image  fields are displayed in one column as intended

SharkyKZ committed:
ModuleHelper prepared statements (#26015)

Hackwar committed:
[4.0] Adding further tests for Joomla\CMS\Table\Table::bind() (#25761)

* Adding further tests for Joomla\CMS\Table\Table::bind()

* Improving json encoded fields in Table::bind()

* Adding more tests for Table::bind() and reset()

* Fixing char error

* Cleanup and fixing of event testing

* Fixing Postgres code

* Updating joomla/database

SharkyKZ committed:
Simplify finder code (#26010)

SharkyKZ committed:
[4.0] Component and Plugin libraries prepared statements (#26012)

C-Lodder committed:
Update body.php (#26035)

SharkyKZ committed:
Use setLimit() to set query limit (#25981)

brianteeman committed:
[4.0] com_users column heading text (#26039)

Simplify / correct the english

brianteeman committed:
plg_content_navigation text (#26038)

Simplify the english

SharkyKZ committed:
LibraryHelper prepared statements (#26031)

SharkyKZ committed:
CMSHelper prepared statements (#26032)

brianteeman committed:
[4.0] Permissions widths (#26027)

* [4.0] Permissions widths

There is no specific width set on the columns in the permissions table. As a result as you move between access levels you will see the columns jump around depending on the width of the content on the first row in the last column.

This PR resolves this by setting explicit widths for the columns

brianteeman committed:
[4.0] Article Options (#26043)

* [4.0] Article Options

Remove the Global fieldset and move those fields to the Options fieldset

Global makes no sense and they're no different really from the Options

SharkyKZ committed:
UserGroupsHelper prepared statements (#26030)

twister65 committed:
[4.0] Get the dbtype from the DatabaseDriver Instance (#25729)

infograf768 committed:
[4.0][Admin menu] Allow multiple menus including expanded ones in sidebar (#25995)

HLeithner committed:
[plg_sampledata_multilang] Convert to prepared statement (#25433)

SharkyKZ committed:
[4.0] ContentHelper prepared statements (#26029)

brianteeman committed:
[4.0] Search button in the toolbar (#26042)

Removes the css that changed the colour of the search button in the toolbar - it is not consistent with the blue used elsewhere in the toolbar such as list and direction

Commit:
Fix language constant (#26049)

infograf768 committed:
[4.0][Admin template] Menu Dashboard: Display language badge for menus containing home (#25944)

chmst committed:
fix unpublish submenu module (#26009)

dgrammatiko committed:
Merge branch '4.0-dev' into spinner-nth-attempt

wilsonge committed:
Merge pull request #25776 from dgrammatiko/spinner-nth-attempt

[4.0] Spinner nth attempt

SharkyKZ committed:
Prepared statements (#26059)

Commit:
Merge remote-tracking branch 'origin/4.0-dev' into editor

# Conflicts:
#	.drone.yml
#	administrator/components/com_banners/tmpl/banner/edit.php
#	administrator/components/com_checkin/Controller/DisplayController.php
#	administrator/components/com_config/tmpl/application/default_filters.php
#	administrator/components/com_config/tmpl/application/default_permissions.php
#	administrator/components/com_config/tmpl/component/default.php
#	administrator/components/com_content/forms/article.xml
#	administrator/components/com_installer/Controller/DatabaseController.php
#	administrator/components/com_installer/Controller/DiscoverController.php
#	administrator/components/com_installer/Controller/DisplayController.php
#	administrator/components/com_installer/Controller/UpdateController.php
#	administrator/components/com_joomlaupdate/Controller/DisplayController.php
#	administrator/components/com_menus/presets/joomla.xml
#	administrator/components/com_menus/presets/menus.xml
#	administrator/components/com_postinstall/Controller/DisplayController.php
#	administrator/components/com_tags/tmpl/tag/edit.php
#	administrator/language/en-GB/en-GB.com_cpanel.ini
#	administrator/language/en-GB/en-GB.mod_menu.ini
#	administrator/language/en-GB/en-GB.mod_submenu.ini
#	administrator/language/en-GB/en-GB.mod_submenu.sys.ini
#	administrator/modules/mod_frontend/mod_frontend.xml
#	administrator/modules/mod_menu/tmpl/default.php
#	administrator/modules/mod_menu/tmpl/default_submenu.php
#	administrator/modules/mod_privacy_status/helper.php
#	administrator/modules/mod_submenu/Menu/Menu.php
#	administrator/modules/mod_submenu/mod_submenu.php
#	administrator/modules/mod_submenu/mod_submenu.xml
#	administrator/modules/mod_submenu/tmpl/default.php
#	administrator/templates/atum/error_full.php
#	administrator/templates/atum/error_login.php
#	administrator/templates/atum/index.php
#	administrator/templates/atum/language/en-GB/en-GB.tpl_atum.ini
#	administrator/templates/atum/login.php
#	administrator/templates/atum/scss/_variables.scss
#	administrator/templates/atum/scss/blocks/_form.scss
#	administrator/templates/atum/scss/blocks/_header.scss
#	administrator/templates/atum/scss/blocks/_login.scss
#	administrator/templates/atum/scss/blocks/_searchtools.scss
#	administrator/templates/atum/scss/blocks/_sidebar.scss
#	administrator/templates/atum/scss/blocks/_toolbar.scss
#	administrator/templates/atum/scss/pages/_com_tags.scss
#	administrator/templates/atum/scss/template-rtl.scss
#	build/media_source/templates/atum/js/template.es6.js
#	composer.lock
#	package-lock.json

bahl24 committed:
try to fix npm drone err

Commit:
Update package-lock file to fix npm error

Commit:
Set drone signature from branch development
